src/Compilers/CSharp/Portable/Symbols/TypedConstantExtensions.cs

```diff
@@ -213,7 +214,10 @@ private static string DisplaySignedEnumConstant(TypedConstant constant, SpecialT
             }
 
             // Unable to decode the enum constant, just display the integral value
-            return constant.ValueInternal.ToString();
+            Debug.Assert(constant.ValueInternal is object);
```

Comment:
>Debug.Assert(constant.ValueInternal is object); [](start = 12, length = 47)

Why assert rather than using `!`?

Reply:
Same rationale

---
In reply to: [443064067](https://github.com/dotnet/roslyn/pull/44266#discussion_r443064067) [](ancestors = 443064067)